packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -109,76 +54,430 @@ const invalidateCustomerBasketsQuery = (
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    updateGiftCertificateItemInBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    updateItemInBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    updatePaymentInstrumentInBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                {queryKey: getPaymentMethodsForBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
+    },
+    updateShipmentForBasket(customerId, {parameters}, response) {
+        return {
+            update: [
+                {queryKey: getBasket.queryKey(parameters)},
+                ...(customerId
+                    ? [
+                          {
+                              queryKey: getCustomerBaskets.queryKey({...parameters, customerId}),
+                              updater: (oldData: CustomerBasketsResult | undefined) =>
+                                  customerBasketsUpdater(parameters, response, oldData)
+                          }
+                      ]
+                    : [])
+            ]
+        }
```

Comment:
Seems like there's a bunch of methods that have this implementation; can they be extracted to a helper?

Reply:
Yep, I was thinking the same. I extracted the code to `updateCustomerBasketsQuery` helper method.